sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuiteBase.scala

```diff
@@ -126,4 +127,512 @@ trait ShowTablesSuiteBase extends QueryTest with DDLCommandTestUtils {
+    }
+  }
+
+  test("show table extended in non-partitioned table") {
+    val namespace = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(namespace, table, catalog) { tbl =>
+      sql(s"CREATE TABLE $tbl (id bigint, data string) $defaultUsing")
+      val e = intercept[AnalysisException] {
+        sql(s"SHOW TABLE EXTENDED IN $catalog.$namespace LIKE '$table' PARTITION(id = 1)")
+      }
+      checkError(
+        exception = e,
+        errorClass = e.getErrorClass,
+        parameters = e.getErrorClass match {
+          case "_LEGACY_ERROR_TEMP_1251" =>
+            Map("action" -> "SHOW TABLE EXTENDED", "tableName" -> table) // v1 v2
+          case "_LEGACY_ERROR_TEMP_1231" =>
+            Map("key" -> "id", "tblName" -> s"`$catalog`.`$namespace`.`$table`") // hive
+        }
+      )
+    }
+  }
+
+  test("show table extended in a not existing partition") {
+    val namespace = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(namespace, table, catalog) { tbl =>
+      sql(s"CREATE TABLE $tbl (id bigint, data string) $defaultUsing PARTITIONED BY (id)")
+      sql(s"ALTER TABLE $tbl ADD PARTITION (id = 1)")
+      checkError(
+        exception = intercept[AnalysisException] {
+          sql(s"SHOW TABLE EXTENDED IN $catalog.$namespace LIKE '$table' PARTITION(id = 2)")
+        },
+        errorClass = "PARTITIONS_NOT_FOUND",
+        parameters = Map(
+          "partitionList" -> "PARTITION (`id` = 2)",
+          "tableName" -> "`ns1`.`tbl`"
+        )
+      )
+    }
+  }
+
+  test("show table extended in multi partition key table") {
+    val namespace = "ns1"
+    val table = "tbl"
+    withNamespaceAndTable(namespace, table, catalog) { tbl =>
+      sql(s"CREATE TABLE $tbl (id1 bigint, id2 bigint, data string) " +
+        s"$defaultUsing PARTITIONED BY (id1, id2)")
+      sql(s"ALTER TABLE $tbl ADD PARTITION (id1 = 1, id2 = 2)")
+
+      val result = sql(s"SHOW TABLE EXTENDED FROM $catalog.$namespace " +
+        s"LIKE '$table' PARTITION(id1 = 1, id2 = 2)")
+      assert(result.schema.fieldNames ===
+        Seq("namespace", "tableName", "isTemporary", "information"))
+      assert(result.collect()(0).length == 4)
+      assert(result.collect()(0)(0) === namespace)
+      assert(result.collect()(0)(1) === table)
+      assert(result.collect()(0)(2) === false)
+      val actualResult = exclude(result.collect()(0)(3).toString)
+      val expectedResult_v1_v2 = "Partition Values: [id1=1, id2=2]"
+      val expectedResult_hive =
+        """Partition Values: [id1=1, id2=2]
+          |Serde Library: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
+          |InputFormat: org.apache.hadoop.mapred.TextInputFormat
+          |OutputFormat: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
+          |Storage Properties: [serialization.format=1]""".stripMargin
+      assert(actualResult === expectedResult_v1_v2 || actualResult === expectedResult_hive)
+
+      checkError(
```

Comment:
Not clear what does this checks, and how it belongs to the test. Could you put it to a separate test with proper title.

Reply:
Done.